### ISSUE TITLE
Game crashing when editing ANYTHING - TWW3 

### ISSUE THREAD

**Issue by F8XCutie** [Frodo45127/rpfm#195]
Seeing priors posts, 3.0.3 says to have updated and fixed the issue with the game crashing when editing things using rpfm. I may be doing it wrong but whenever I save the 'data' file the game crashes on start-up.

Before the updates it used to work and I could mess around with specifically the 'technology-effects' but now when trying to change that or more it such as things in the 'main unit table' or 'campaign difficulty handicap' it just crashes.

Sorry in advance if the issue is resolved already

**Comment by JimLovesPineapples** [comment]
Same here, no problem before 3.0 but now I can't launch the game. Edited campaign diff handicap

**Comment by Vavohu** [comment]
Did you start with a clean data pack, change to Warhammer 3 game and enable CA editing?
I had issues with 3.0.2 but am currently running multiple mods including both tables and script changes without issues with 3.0.3

**Comment by F8XCutie** [comment]
> Did you start with a clean data pack, change to Warhammer 3 game and enable CA editing? I had issues with 3.0.2 but am currently running multiple mods including both tables and script changes without issues with 3.0.3

Yeah, I even went as far as buying it on steam to see if xbox was the issue. I was using the $1 pass (little side note. for those wanting to test it you should use the Xbox $1 pass)

I enabled CA editing and changed the game to Warhammer 3. It's just randomly stopped working for me and just crashes my game now. Was super excited to see heaps of AI armies and huge battles

**Comment by ahamm3rs** [comment]
> Did you start with a clean data pack, change to Warhammer 3 game and enable CA editing? I had issues with 3.0.2 but am currently running multiple mods including both tables and script changes without issues with 3.0.3

Could you please further describe, how u do it? Did you create mod-files or modified the data.pack (etc.)? In the first case - how did u enable the mods? Via user.script (mod "modxyz.pack";)?

**Comment by Vavohu** [comment]
Could you please further describe, how u do it? Did you create mod-files or modified the data.pack (etc.)? In the first case - how did u enable the mods? Via user.script (mod "modxyz.pack";)?


Update rpfm
Restart rpfm
Have a fresh (reverified) data.pack
Create a backup elsewhere
Open in rpfm with game set to Warhammer 3, CA mods enabled in preferences
Change several values in several different tables under db
Change values in Lua for chaos realm invasion parameters
Save packfile (as root - the default; not as a mod)
Run game

**Comment by Frodo45127** [comment]
First, editing data is NOT RECOMMENDED.

Second, 3.0.3 doesn't fix already broken data.pack files. You need to validate steam cache or equivalent to fix them. 3.0.3 just have the fix to not break them anymore.

Third: the fix REQUIRES BOTH, 3.0.3 AND THE SCHEMA UPDATE FROM YESTERDAY. With both of those, opening and saving data.pack saves a identical packfile, byte by byte (except for the 4 bytes of the timestamp in the header, those have always changed). And yes, I checked the whole 6GB thing for differences.

So basically, avoid editing data.pack as much as possible, and if you edit it, make sure you have both, 3.0.3 and your schemas updated.

**Comment by F8XCutie** [comment]
> Could you please further describe, how u do it? Did you create mod-files or modified the data.pack (etc.)? In the first case - how did u enable the mods? Via user.script (mod "modxyz.pack";)?

Yeah I would love to know how to go about doing this too. I validated all my files and used data.pack and just deleted things I didn't need then I 'saved as' a 'mod'. Then went to preferences.scirpt and put in mod “Kislev.pack”;

That didn't work for me. I don't know what I'm doing wrong here. Thank you in advance is someone can help

**Comment by frstwlf** [comment]
I fixed the issue and managed to modify the data.pack by deleting the rpfm folder from appdata and NOT generating the dependencies.

Thank you Frodo45127 for all your hard work!

**Comment by ahamm3rs** [comment]
RPFM is working here too. I did as Vavohu described - new data.pack etc - working fine now.
Btw - i turned steam to offline-mode, so no CA-Update crash this working and modified game now 👯 

Thanks alot @ Frodo45127 :-)

**Comment by F8XCutie** [comment]
Alright, issue has been resolved on my end. I had to do what @frstwlf said and it works out fine.

Thank you heaps everyone!

**Comment by JimLovesPineapples** [comment]
> I fixed the issue and managed to modify the data.pack by deleting the rpfm folder from appdata and NOT generating the dependencies.
> 
> Thank you Frodo45127 for all your hard work!

Deleting appdata also worked for me

**Comment by Frodo45127** [comment]
Deleting appdata works because you're forcing RPFM to download the updated schemas after that. Which you can just download. Anyway, closing this issue, as the incorrect encoding issues are fixed.
